narwhal/consensus/src/tests/bullshark_tests.rs

```diff
@@ -818,3 +822,257 @@ async fn restart_with_new_committee() {
         handle.await.unwrap();
     }
 }
+
+/// The test ensures the following things:
+/// * garbage collection is removing the certificates from lower rounds according to gc depth
+/// * we garbage collect/delete certificates bellow the last commit round immediately once we commit them
+/// * no certificate will ever get committed past the gc round
+/// * existing uncommitted certificates in DAG (ex from slow nodes where no-one references them) they
+/// get cleaned up.
```

Comment:
Not sure if we already have it but we should add this as one of our e2e tests. Where if we have uncommitted certs that are removed due to GC, the batches get rebroadcast and eventually get committed.

Reply:
Good idea @arun-koshy ! A bit more challenging as e2e as this would really need to pick into the server internals - but we can definitely have a look on this flow e2e tested